src/pydvl/utils/parallel/map_reduce.py

```diff
@@ -175,24 +158,21 @@ def __init__(
 
     def __call__(
         self,
-        inputs: Union[Collection[T], Any],
-        *,
-        n_jobs: Optional[int] = None,
-        n_runs: Optional[int] = None,
-        chunkify_inputs: Optional[bool] = None,
+        inputs: Union[Sequence[T], Sequence[Utility], Utility],
     ) -> List[R]:
-        if n_jobs is not None:
-            self.n_jobs = n_jobs
-        if n_runs is not None:
-            self.n_runs = n_runs
-        if chunkify_inputs is not None:
-            self.chunkify_inputs = chunkify_inputs
-
-        map_results = self.map(inputs)
+        if isinstance(inputs, Utility):
+            inputs_ = self.parallel_backend.put(inputs)
+        elif len(inputs) > 0 and isinstance(inputs[0], Utility):
+            inputs_ = [self.parallel_backend.put(x) for x in inputs]
+        else:
+            inputs_ = inputs
```

Comment:
Why can only `put()` utility objects? Shouldn't this be the same for any type? I'd expect to have to put heavy np arrays to the store as well, right?

Reply:
I agree, but I couldn't find a way to easily do this.

I have working on some changes to this locally while also fixing type hints. I will push them and ask for a feedback again.